api/src/main/java/org/openmrs/module/fhir2/api/dao/impl/BaseDaoImpl.java

```diff
@@ -440,7 +440,7 @@ protected boolean containsAlias(Iterator<CriteriaImpl.Subcriteria> subcriteriaIt
 			}
```

Comment:
We should also address the `Optional.empty()` above this line.

Reply:
I have a couple of questions here, 

1. The code `OTHER` has a mapping to `'O'`, as shown in the `GenderTranslatorImplTest`. Keeping that in mind, should we change the Criterion to match `'O'` specifically? Because the code `UNKNOWN` too has a mapping to `'U'`.

2. Should `NULL` match any resource or those which have gender set to empty or null (if that's even possible)?

Reply:
So remember what we're mapping to here: we're querying the OpenMRS database for people matching whatever gender. In OpenMRS, there are really just three possible gender codes `M`, `F` and `null`.  Maybe a better way to support this is to map all of `OTHER`, `UNKNOWN`, and `NULL` to `gender is null`. We should probably fix the gender translator to reflect this as well. I suppose it should just return `M` for "male", `F` for "female" and `null` for "unknown".